_search-plugins/search-relevance/rerank-by-field-cross-encoder.md

```diff
@@ -0,0 +1,276 @@
+    - The `text_pair` field dynamically retrieves the search query text (`multi_match.query`) from the search request. 
+
+    The combination of `text` (document `facts`) and `text_pair` (search `query`) allows the cross-encoder model to compare the relevance of the document to the query, considering their semantic relationship.
+
+- The `output_map` specifies how the output of the model is mapped to fields in the response:
+    - The `rank_score` field in the response will store the model's relevance score which will be used to perform reranking.
+    
+When using the `by_field` rerank type, the `rank_score` field will contain the same score as the `_score` field. To remove the `rank_score` field from the search results, set `remove_target_field` to `true`. The original BM25 score, before reranking, is included for debugging purposes by setting `keep_previous_score` to `true`. This allows you to compare the original score with the reranked score to evaluate improvements in search relevance.
+    
+To create the search pipeline, send the following request:
+
+```json
+PUT /_search/pipeline/my_pipeline
+{
+  "response_processors": [
+    {
+      "ml_inference": {
+        "tag": "ml_inference",
+        "description": "This processor runs ml inference during search response",
+        "model_id": "<model_id_from_step_3>",
+        "function_name": "REMOTE",
+        "input_map": [
+          {
+            "text": "facts",
+            "text_pair":"$._request.query.multi_match.query"
+          }
+        ],
+        "output_map": [
+          {
+            "rank_score": "$.score"
+          }
+        ],
+        "full_response_path": false,
+        "model_config": {},
+        "ignore_missing": false,
+        "ignore_failure": false,
+        "one_to_one": true
+      },
+       
+      "rerank": {
+        "by_field": {
+          "target_field": "rank_score",
+          "remove_target_field": true,
+          "keep_previous_score" : true
+          }
+      }
+    
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+## Step 5: Search using reranking
+
+Use the following request to search indexed documents and rerank them using the cross-encoder model. The request retrieves documents containing any of the specified terms in the `description` or `facts` fields. These terms are then used to compare and rerank the matched documents:
+
+```json
+POST /nyc_areas/_search?search_pipeline=my_pipeline
+{
+  "query": {
+    "multi_match": {
+      "query": "artists art creative community",
+      "fields": ["description", "facts"]
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+In the response, the `previous_score` field contains the document's BM25 score, which it would have received if you hadn't applied the pipeline. Note that while BM25 ranked Astoria as the highest, the cross-encoder prioritized Harlem because it matched more search terms:
```

Comment:
```suggestion
In the response, the `previous_score` field contains the document's BM25 score, which it would have received if you hadn't applied the pipeline. Note that while BM25 ranked "Astoria" the highest, the cross-encoder model prioritized "Harlem" because it matched more search terms:
```